### PR TITLE
(DOCSP-38560): Edge Server Admin API Endpoint Documentation

### PR DESCRIPTION
**Pull request by dacharyc**
## Pull Request Info

Jira ticket: https://jira.mongodb.org/browse/DOCSP-38560

- [Admin API](https://docs-atlas-staging.mongodb.com/atlas-app-services/docsworker-xlarge/DOCSP-38560/admin/api/v3/#tag/edge-server): New Edge Server endpoints:
  - [List Edge Server instances](https://docs-atlas-staging.mongodb.com/atlas-app-services/docsworker-xlarge/DOCSP-38560/admin/api/v3/#tag/edge-server/operation/adminGetAllEdgeServerInstances)
  - [Get information about an Edge Server instance](https://docs-atlas-staging.mongodb.com/atlas-app-services/docsworker-xlarge/DOCSP-38560/admin/api/v3/#tag/edge-server/operation/adminGetEdgeServerInstance)
  - [Create an Edge Server instance](https://docs-atlas-staging.mongodb.com/atlas-app-services/docsworker-xlarge/DOCSP-38560/admin/api/v3/#tag/edge-server/operation/adminCreateEdgeServerInstance)
  - [Delete an Edge Server instance](https://docs-atlas-staging.mongodb.com/atlas-app-services/docsworker-xlarge/DOCSP-38560/admin/api/v3/#tag/edge-server/operation/adminDeleteEdgeServerInstance)
  - [Regenerate an Edge Server registration key](https://docs-atlas-staging.mongodb.com/atlas-app-services/docsworker-xlarge/DOCSP-38560/admin/api/v3/#tag/edge-server/operation/adminRegenerateEdgeServerRegistrationKey)
  - [Rename an Edge Server instance](https://docs-atlas-staging.mongodb.com/atlas-app-services/docsworker-xlarge/DOCSP-38560/admin/api/v3/#tag/edge-server/operation/adminRenameEdgeServerInstance)

A few endpoints aren't testable right now because of bugs or because they're not yet in prod, so adding these todo items which should carry over to the feature branch:

TODOS for Feature Branch:
- [ ] Test Delete API
- [ ] Test Regenerate API
- [ ] Test Rename API
- [ ] Populate data for Get Edge Server Info/metrics object

### Release Notes

- **Admin API**
  - New Edge Server endpoints:
    - List Edge Server instances
    - Get information about an Edge Server instance
    - Create an Edge Server instance
    - Delete an Edge Server instance
    - Regenerate an Edge Server registration key
    - Rename an Edge Server instance

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-app-services/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/openapi-admin-v3.yaml [hunk 1]
```diff
@@ -4387,6 +4387,137 @@ paths:
...
+        required: true
+        content:
+            application/json:
+              schema:
+                type: object
+                required: ["name"]
+                properties:
+                  name:
+                    type: string
+                    example: Edge-Store-42
```
Comment (sudssm) [inline source/openapi-admin-v3.yaml:4432]: idk if its worth noting that this must be unique

File: source/openapi-admin-v3.yaml [hunk 2]
```diff
@@ -5632,6 +5771,157 @@ components:
...
+        details about metrics when available.]
+    EdgeServerCreateResponse:
+      type: object
+      properties:
+        id:
+          type: string
+          example: 6622b8f5118368b943d228a9
+          description: |-
+            The unique identifier of the Edge Server instance, of type 
+            ObjectId, as a string. You can use this identifier when configuring
```
Comment (sudssm) [inline source/openapi-admin-v3.yaml:5910]: actually, you wouldn't use the edge server ID to configure permissions; you'd use it to delete/ update the edge server. you'duse the user id to configure perms

File: source/openapi-admin-v3.yaml [hunk 3]
```diff
@@ -5632,6 +5771,157 @@ components:
...
+            permissions.
+        user_id:
+          type: string
+          example: 6622b8f5118368b943d2288a
+          description: |-
+            The Edge Server itself is a user. This is the unique identifier of
```
Comment (sudssm) [inline source/openapi-admin-v3.yaml:5916]: you'd use this to configure permissions

File: source/openapi-admin-v3.yaml [hunk 4]
```diff
@@ -5632,6 +5771,157 @@ components:
...
+          description: |-
+            The Edge Server itself is a user. This is the unique identifier of 
+            the Edge Server user, of type ObjectId, as a string.
+        registration_key:
+          type: string
+          example: Qi4GgpvVduwwt4k1w77ZzaQ1n4SdTWC8rLBD9vyTib1w9G3CX2ta8WrJvt3giEK7
+          description: |-
+            Save this information. When you install the Edge Server code on the
+            host, enter this registration key. The Edge Server exchanges this 
+            for an API Key that it uses to authenticate with Atlas.
```
Comment (sudssm) [inline source/openapi-admin-v3.yaml:5924]: worth adding that it can only be used once?

File: source/openapi-admin-v3.yaml [hunk 5]
```diff
@@ -5632,6 +5771,157 @@ components:
...
+          type: string
+          enum:
+            - "active"
+            - "deleted"
+            - "failed"
+            - "registered"
+            - "unregistered"
+          description: |-
+            The current status of the Edge Server instance.
+        hostname:
```
Comment (sudssm) [inline source/openapi-admin-v3.yaml:5858]: this is not going to be populated for the foreseeable future; we should just omit it

File: source/openapi-admin-v3.yaml [hunk 6]
```diff
@@ -4387,6 +4387,137 @@ paths:
...
+    parameters:
+      - "$ref": "#/components/parameters/GroupId"
+      - "$ref": "#/components/parameters/AppId"
+      - "$ref": "#/components/parameters/EdgeServerId"
+
+    post:
+      tags:
+        - edge-server
+      operationId: adminRegenerateEdgeServerRegistrationKey
+      summary: Regenerate Edge Server registration key
```
Comment (sudssm) [inline source/openapi-admin-v3.yaml:4483]: we should update all instances of "registration key" with "registration token". except, unfortunately, the endpoint itself which we will keep as-is for now

**Review by sudssm — APPROVED** [review]

File: source/openapi-admin-v3.yaml [hunk 7]
```diff
@@ -4387,6 +4387,137 @@ paths:
...
+        content:
+            application/json:
+              schema:
+                type: object
+                required: ["name"]
+                properties:
+                  name:
+                    type: string
+                    example: Edge-Store-42
+                    description: A human-readable name for the Edge Server instance.
```
Comment (cbullinger) [inline source/openapi-admin-v3.yaml:4433]:
```suggestion
                    description: A unique human-readable name for the Edge Server instance.
```

File: source/openapi-admin-v3.yaml [hunk 8]
```diff
@@ -4387,6 +4387,137 @@ paths:
...
+                    type: array
+                    items:
+                      $ref: "#/components/schemas/EdgeServerInstance"
+
+    post:
+      tags:
+        - edge-server
+      operationId: adminCreateEdgeServerInstance
+      summary: Create an Edge Server instance
+      description: Create a new Edge Server instance.
```
Comment (cbullinger) [inline source/openapi-admin-v3.yaml:4421]: Do we want to mention any of the registration process here? e.g.
```suggestion
      description: Create a new Edge Server instance. The response returns a registration key (token?) that you use to authenticate Edge Server on the host.
```

File: source/openapi-admin-v3.yaml [hunk 9]
```diff
@@ -4387,6 +4387,137 @@ paths:
...
+                type: object
+                required: ["name"]
+                properties:
+                  name:
+                    type: string
+                    example: Edge-Store-42
+                    description: A human-readable name for the Edge Server instance.
+                    required: true
+      responses:
+        "201":
```
Comment (cbullinger) [inline source/openapi-admin-v3.yaml:4436]: I know we're crunched for time on this, but in general, should we be doc'ing the errors too?

**Review by cbullinger — APPROVED** [review]
current draft lgtm. happy to re-review once you apply tech feedback or after additional updates 👍

**Pull request merged**